FR: Deprecation Warning for Enum Access

Per [this comment here](https://github.com/napari/napari/issues/3263#issuecomment-907159978):

> starting with sip v4.19.9 (around PyQt 5.11) all enums are now scoped, and starting in PyQt6, the backwards compatibility allowing enums to be accessed in their top level namespace has been removed

This is confirmed by the [PyQt5 Gotcha's docs](https://www.riverbankcomputing.com/static/Docs/PyQt5/gotchas.html#enums).

Since end users may be accessing different versions of PyQt/PySide through `qtpy`, would it be possible to add a `DeprecationWarning` for the corresponding versions? (most likely this could be determined at runtime via `QT_API`)

e.g. I noticed a `"Type[QEvent]" has no attribute "Gesture"` mypy error in legacy code where `Gesture` was accessed like so:

![image](https://user-images.githubusercontent.com/59346180/176490222-5bd174c3-ba04-40f1-98cc-4aa91b5a6f71.png)

and the fix was instead to use `QtCore.QEvent.Type.Gesture`.

Perhaps something like:

```
DeprecationWarning: Enums are scoped beginning with PyQt 5.11 (and its corresponding PySide 2 version). Access to enums through top-level namespaces has been deprecated beginning in PyQt 6.

Use the full path to the enumeration instead. For example:

QtCore.QEvent.Type.Gesture

instead of 

QtCore.QEvent.Gesture
```

I can confirm (using the `Python` `OUTPUT` tab) when the arguments are passed separately (not comma-delimited), this works for `mypy`:

```
> .\.venv\Scripts\python.exe ~\.vscode\extensions\ms-python.python-2022.8.1\pythonFiles\linter.py -p mypy ./.venv/Scripts/mypy.exe --config-file mypy.ini --always-true=PYSIDE6 --always-false=PYQT5,PYSIDE2,PYQT6 .\pyvistaqt\window.py
cwd: .
##########Linting Output - mypy##########

pyvistaqt\window.py:33: error: "Type[QEvent]" has no attribute "Gesture"
Found 1 error in 1 file (checked 1 source file)

```

as opposed to

```
> .\.venv\Scripts\python.exe ~\.vscode\extensions\ms-python.python-2022.8.1\pythonFiles\linter.py -p mypy ./.venv/Scripts/mypy.exe --config-file mypy.ini --always-true=PYSIDE6 --always-false=PYQT5 --always-false=PYSIDE2 --always-false=PYQT6 .\pyvistaqt\window.py
cwd: .
##########Linting Output - mypy##########

pyvistaqt\window.py:34: error: "Signal" has no attribute "emit"
pyvistaqt\window.py:40: error: "Signal" has no attribute "emit"
Found 2 errors in 1 file (checked 1 source file)
```

and I know how to fix the stubs to solve the error `"Signal" has no attribute "emit"`.

However, I have yet to figure out a way to get this to work for `pylint`. It defaults to using `PySide2`.

--- a/README.md
+++ b/README.md
@@ -71,7 +71,7 @@ conda install qtpy
 ```
 
 
-### Mypy integration
+### Type checker integration
 
 A Command Line Interface (CLI) is offered to help with usage of QtPy.
 Presently, its only feature is to generate command line arguments for Mypy
@@ -100,6 +100,8 @@ the Mypy command line invocation as follows:
 ```bash
 mypy --package mypackage $(qtpy mypy-args)
 ```
+
+For Pyright support and other usage notes, see [this comment](https://github.com/spyder-ide/qtpy/issues/352#issuecomment-1170684412).
 
 
 ## Contributing

